src/main/drivers/bus_spi.c

```diff
@@ -240,6 +240,25 @@ void spiBusSetInstance(busDevice_t *bus, SPI_TypeDef *instance)
     bus->busdev_u.spi.instance = instance;
 }
 
+uint16_t spiCalculateDivider(uint32_t freq)
+{
+#if defined(STM32F4) || defined(STM32G4) || defined(STM32F7)
+    uint32_t spiClk = SystemCoreClock/2;
+#elif defined(STM32H7)
+    uint32_t spiClk = 100000000;
+#endif
+
+    uint16_t divisor = 4;
+
+    spiClk >>= 1;
+
+    for (; spiClk > freq; divisor <<= 1, spiClk >>= 1);
+
+    divisor >>= 1;
```

Comment:
This can be eliminated by starting with `divisor = 2`.

Reply:
Should have spotted that myself. Fixed.